bmx280.c

```diff
@@ -318,13 +407,39 @@ bmx280_t* bmx280_create(i2c_port_t port)
 
         bmx280->i2c_port = port;
         bmx280->slave = 0xDE;
+#else
+bmx280_t* bmx280_create(i2c_master_bus_handle_t bus_handle)
+{
+    bmx280_t* bmx280 = malloc(sizeof(bmx280_t));
+    i2c_config_t dev_cfg = malloc(sizeof(struct i2c_config_t));
+    if (bmx280 && dev_cfg)
+    {
+        memset(bmx280, 0, sizeof(bmx280_t));
+        memset(dev_cfg, 0, sizeof(struct i2c_config_t));
+        bmx280->i2c_cfg = dev_cfg;
+        bmx280->i2c_cfg->bus_handle = bus_handle;
+        bmx280->i2c_cfg->dev_cfg.dev_addr_length = I2C_ADDR_BIT_LEN_7;
+        bmx280->i2c_cfg->dev_cfg.device_address = 0xDE;
+        bmx280->i2c_cfg->dev_cfg.scl_speed_hz =CONFIG_BMX280_I2C_CLK_SPEED_HZ;
+        bmx280->i2c_dev = NULL;
+#endif
         bmx280->chip_id = 0xAD;
     }
+    else
+    {
+        ESP_LOGE("bmx280", "Failed to allocate memory for bmx280.");
+        bmx280_close(bmx280);
+        return NULL;
+    }
     return bmx280;
 }
 
 void bmx280_close(bmx280_t *bmx280)
 {
+    #if CONFIG_USE_I2C_NEW_DEVICE
+    free(bmx280->i2c_cfg);
```

Comment:
I suggest you use an if NULL check around the deconstructor for the properties. It's legal to free(NULL) but not legal to dereference a field.